From: Chris Coulson <chris.coulson@canonical.com>
Date: Mon, 1 Jun 2020 14:03:37 +0100
Subject: UBUNTU: disk/loopback: Don't verify loopback images

When a file is verified, the entire contents of the verified file are
loaded in to memory and retained until the file handle is closed. A
consequence of this is that opening a loopback image can incur a
significant memory cost.

As loopback devices are just another disk implementation, don't treat
loopback images any differently to physical disk images, and skip
verification of them. Files opened from the filesystem within a loopback
image will still be passed to verifier modules where required.

Signed-off-by: Chris Coulson <chris.coulson@canonical.com>
LP: #1878541
Forwarded: yes, https://lists.gnu.org/archive/html/grub-devel/2020-06/msg00002.html
Patch-Name: ubuntu-dont-verify-loopback-images.patch
---
 grub-core/disk/loopback.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/disk/loopback.c b/grub-core/disk/loopback.c
index 4635dcf..679a108 100644
--- a/grub-core/disk/loopback.c
+++ b/grub-core/disk/loopback.c
@@ -102,7 +102,7 @@ grub_cmd_loopback (grub_extcmd_context_t ctxt, int argc, char **args)
     if (grub_strcmp (newdev->devname, args[0]) == 0)
       return grub_error (GRUB_ERR_BAD_ARGUMENT, "device name already exists");
 
-  file = grub_file_open (args[1], type);
+  file = grub_file_open (args[1], type | GRUB_FILE_TYPE_SKIP_SIGNATURE);
   if (! file)
     return grub_errno;
 
